app/src/backend/CalculationService.ts

```diff
@@ -73,70 +81,193 @@ export default class CalculationService {
-            throw new createHttpError.BadRequest(
-              `Emissions factor ${emissionsFactor.id} has no emissions per activity`,
-            );
-          }
-          // this rounds/ truncates!
-          const amount = BigInt(
-            activityAmount * emissionsFactor.emissionsPerActivity,
-          );
-          const { co2eq, co2eqYears } = this.calculateCO2eq(
+        gases =
+          CalculationService.handleActivityAmountTimesEmissionsFactorFormula(
+            activityValue,
             gasToCO2Eqs,
-            gasValue.gas!,
-            amount,
           );
-          totalCO2e += co2eq;
-          totalCO2eYears = Math.max(totalCO2eYears, co2eqYears);
-
-          return { gas: gasValue.gas!, amount };
-        });
         break;
+      case "methane-commitment":
+        gases = CalculationService.handleMethaneCommitmentFormula(
+          activityValue,
+          gasToCO2Eqs,
+        );
+        break;
+      case "induced-activity-1": // TODO or VKT-1? What is it in the hierarchy.json file?
+        gases = CalculationService.handleVkt1Formula(
+          activityValue,
+          gasToCO2Eqs,
+        );
       default:
         throw new createHttpError.NotImplemented(
           `Formula ${formula} not yet implemented for input methodology ${inventoryValue.inputMethodology}`,
         );
     }
+
+    for (const gas of gases) {
+      const { co2eq, co2eqYears } = this.calculateCO2eq(
+        gasToCO2Eqs,
+        gas.gas,
+        gas.amount,
+      );
+      totalCO2e += co2eq;
+      totalCO2eYears = Math.max(co2eqYears, totalCO2eYears);
+    }
+
     return {
       totalCO2e,
       totalCO2eYears,
       gases,
     };
   }
+
+  static handleVkt1Formula(
+    activityValue: ActivityValue,
+    gasToCO2Eqs: GasToCO2Eq[],
+  ): Gas[] {
+    const data = activityValue.activityData;
+    if (!data) {
+      throw new createHttpError.BadRequest(
+        "Activity has no data associated, so it can't use the formula",
+      );
+    }
+
+    const gases = activityValue.gasValues.map((gasValue) => {
+      if (!gasValue.gas) {
+        throw new createHttpError.BadRequest(
+          "Activity has a GasValue with no `gas` name",
+        );
+      }
+      const emissionsFactor = gasValue.emissionsFactor;
+      const emissions =
+        data["activity-value"] *
+        data["intensity"] *
+        (emissionsFactor.emissionsPerActivity ?? 0); // TODO throw BadRequest error if no EF present?
+      return { gas: gasValue.gas, amount: BigInt(emissions) };
+    });
+
+    return gases;
+  }
+
+  static handleMethaneCommitmentFormula(
+    activityValue: ActivityValue,
+    gasToCO2Eqs: GasToCO2Eq[],
+  ): Gas[] {
+    const data = activityValue.activityData;
+    if (!data) {
+      throw new createHttpError.BadRequest(
+        "Activity has no data associated, so it can't use the formula",
+      );
+    }
+    const foodFraction = (data["food-fraction"] || 0) / 100.0;
```

Comment:
a more DRY approach could be

const getFraction = (key) => (data[key] || 0) / 100.0;

[foodFraction, gardenWasteFraction, paperFraction, woodFraction, textilesFraction, industrialWasteFraction] = ["food-fraction", "garden-waste-fraction",
"paper-fraction", "wood-fraction", "textiles-fraction", "industrial-waste-fraction"].map(getFraction)